### DISCUSSION TITLE
cell_id undetected by NWBInspector

### DISCUSSION THREAD

**Discussion by Cryon1x** [NeurodataWithoutBorders/helpdesk#54] [General]
I am converting abf files to nwb using neuroconv. When I test the new nwb files, though I have manually added the "cell_id" field to each electrode, it still throws an error of Message: Please include a unique cell_id associated with this IntracellularElectrode. 

Here is my metadata: 

``` 
DeepDict: {'NWBFile': {'session_description': 'Auto-generated by neuroconv', 'identifier': '318a1201-0588-462a-b24c-66cef327dfcb', 'session_start_time': datetime.datetime(2021, 10, 12, 16, 0, 8)}, 'Icephys': {'Device': [{'name': 'DeviceIcephys', 'description': 'no description'}], 'Electrode': [{'name': 'electrode-0', 'description': 'no description', 'device': 'DeviceIcephys'}, {'name': 'electrode-1', 'description': 'no description', 'device': 'DeviceIcephys'}], 'Sessions': [{'name': 'afterKA15min_0000.abf', 'relative_session_start_time': 0.0, 'icephys_experiment_type': None, 'stimulus_type': 'not described', 'recordings': [{'intracellular_recordings_table_ind': 0, 'simultaneous_recordings_table_ind': 0, 'sequential_recordings_table_ind': 0}, {'intracellular_recordings_table_ind': 1, 'simultaneous_recordings_table_ind': 0, 'sequential_recordings_table_ind': 0}]}], 'cell_id': 'ID001', 'Electrodes': [{'name': 'icephys_electrode_0', 'description': 'no description', 'device_name': 'DeviceIcephys', 'cell_id': 'ID001'}, {'name': 'icephys_electrode_1', 'description': 'no description', 'device_name': 'DeviceIcephys', 'cell_id': 'ID002'}]}, 'icephys_electrode_1': {}, 'icephys_electrode_41': {}, 'icephys_electrodffdse_1': {}, 'Electrode': {}, 'Electrodes': {}}
 ```

Am I doing something wrong?

**Comment by CodyCBakerPhD** [comment]
Hi there,

Are you able to share the file causing the problem? And also the NeuroConv code that was used to create it?

**Comment by Cryon1x** [comment]
Unfortunately my file is too large (30 MB). Should we take this offline? 

I have enclosed my code below: 

```
interface = AbfInterface(
    ...:     file_paths=["afterKA15min_0000.abf"],
    ...: )


metadata = interface.get_metadata()

metadata['Icephys']['Electrode'][0].update(cell_id = "ID001")
metadata['Icephys']['Electrode'][1].update(cell_id = "ID002")

interface.run_conversion(nwbfile_path="output.nwb", metadata=metadata)
```

**Comment by CodyCBakerPhD** [comment]
@Cryon1x

> Unfortunately my file is too large (30 MB

I've created a Google Drive that you can upload the file to here: https://drive.google.com/drive/folders/1l-0nrIeprUYyCffV1LGnFObcL3mwLqdV?usp=sharing

Being able to see the contents of the file and run the NWB Inspector myself will help speed up debugging by a lot

**Comment by Cryon1x** [comment]
uploaded. Please let me know what I'm doing wrong at your earliest convenience. Greatly appreciate it.

**Comment by CodyCBakerPhD** [comment]
Oh, thanks for including the source data too! That will help if it's a NeuroConv problem

Will investigate and get back to you soon~

**Comment by CodyCBakerPhD** [comment]
@Cryon1x

OK I think I've fixed the issue - that metadata was not getting propagated down to the electrode creator; there was also a mismatch in the metadata schema definition

Can you try running it from the following dev branch?

```
pip install git+https://github.com/catalystneuro/neuroconv.git@abf_electrode_fixes
```

(you may need to install git, `conda install git` if using a conda environment)

You will also need to update your script to

```python
from neuroconv.datainterfaces import AbfInterface

interface = AbfInterface(
    file_paths=["afterKA15min_0000.abf"],
)


metadata = interface.get_metadata()

metadata['Icephys']['Electrodes'][0].update(cell_id = "ID001")
metadata['Icephys']['Electrodes'][1].update(cell_id = "ID002")

interface.run_conversion(nwbfile_path="reproduced_output.nwb", metadata=metadata, overwrite=True)
```

**Comment by Cryon1x** [comment]
Issue is fixed using that branch! Thank you so much. 

Would this branch be merged with main anytime soon? I have colleagues who need to publish soon.

**Comment by CodyCBakerPhD** [comment]
Depends on the timescale implied by 'soon' - if @luiztauffer (original creator of that interface and tools) or someone else on the team can give a review of https://github.com/catalystneuro/neuroconv/pull/538 within the next couple of days then yes I can merge and cut a new bug fix release very quickly

**Comment by CodyCBakerPhD** [comment]
@rly or @oruebel can one of you transfer this to the official NWB helpdesk and also convert to a discussion? I don't have permissions

Thanks

Also @Cryon1x if you ever find any more problems with NeuroConv specifically, feel free to raise an issue directly over there: https://github.com/catalystneuro/neuroconv/issues

Or for more general questions about any part of the NWB ecosystem, check out https://github.com/NeurodataWithoutBorders/helpdesk/discussions

**Comment by CodyCBakerPhD** [comment]
@Cryon1x The NeuroConv patch was approved and released over the weekend 👍
